external/meta-python/recipes-devtools/python3-tensorrt/python3-tensorrt_7.1.3-1.bb

```diff
@@ -0,0 +1,56 @@
+SUMMARY = "Tensorrt Python bindings"
+
+LICENSE = "Proprietary"
+LIC_FILES_CHKSUM = "file://python/packaging/LICENSE.txt;md5=c291e0a531e08d4914e269730ba2f70d"
+
+DEPENDS = "python3-pybind11 tensorrt"
+
+inherit setuptools3 cmake pkgconfig cuda
+
+SRC_REPO = "github.com/NVIDIA/TensorRT.git;protocol=https"
+SRCBRANCH = "master"
+SRC_URI = "git://${SRC_REPO};branch=${SRCBRANCH} \
+           file://0001-Do-not-overwrite-CMAKE_CXX_FLAGS.patch \
+           file://0002-pyOnnx-fixups-for-TensorRT-7.1-API.patch \
+           "
+SRCREV = "ab20a8ac26da03df5e434b4f9d1da0156ca019b6"
+
+S = "${WORKDIR}/git"
+
+OECMAKE_SOURCEPATH = "${S}/python"
+DISTUTILS_SETUP_PATH = "${B}"
+
+EXTRA_OECMAKE = "-DONNX_INC_DIR=${STAGING_INCDIR} -DPYBIND11_DIR=${STAGING_DIR_TARGET} \
+                 -DTARGET_ARCHITECTURE=${HOST_ARCH} -DCMAKE_BUILD_TYPE=Release \
+                 -DPY_INCLUDE=${STAGING_INCDIR}/${PYTHON_DIR}"
+
+CXXFLAGS += "${CUDA_CXXFLAGS}"
+
+do_configure_append() {
+    TRT_MAJOR=$(awk '/^\#define NV_TENSORRT_MAJOR/ {print $3}' ${STAGING_INCDIR}/NvInferVersion.h)
+    TRT_MINOR=$(awk '/^\#define NV_TENSORRT_MINOR/ {print $3}' ${STAGING_INCDIR}/NvInferVersion.h)
+    TRT_PATCH=$(awk '/^\#define NV_TENSORRT_PATCH/ {print $3}' ${STAGING_INCDIR}/NvInferVersion.h)
+    TRT_BUILD=$(awk '/^\#define NV_TENSORRT_BUILD/ {print $3}' ${STAGING_INCDIR}/NvInferVersion.h)
+    TRT_VERSION=${TRT_MAJOR}.${TRT_MINOR}.${TRT_PATCH}.${TRT_BUILD}
+    TRT_MAJMINPATCH=${TRT_MAJOR}.${TRT_MINOR}.${TRT_PATCH}
+    varsubst() {
+        sed -e "s|\#\#TENSORRT_VERSION\#\#|${TRT_VERSION}|g" \
+	    -e "s|\#\#TENSORRT_MAJMINPATCH\#\#|${TRT_MAJMINPATCH}|g" $1 >$2
+    }
+
+    [ -d ${B}/tensorrt ] || mkdir ${B}/tensorrt
+    varsubst ${S}/python/packaging/setup.cfg ${B}/setup.cfg
+    varsubst ${S}/python/packaging/setup.py ${B}/setup.py
+    varsubst ${S}/python/packaging/tensorrt/__init__.py ${B}/tensorrt/__init__.py
+    cp ${S}/python/packaging/LICENSE.txt ${B}/tensorrt
+}
+
+do_compile_append() {
+    distutils3_do_compile
```

Comment:
I think that `distutils3_do_compile` is not needed because it will be done by default when inheriting `distutils3`

Reply:
I've reworked each of the task definitions to explicitly call on the relevant functions from the inherited classes in the right order.  I have left in the distutils3_do_compile function because it does actually do something.  Yes, distutils3_do_install will also execute the compile tasks, but it does so under pseudo context, which could potentially cause some issues.